neptune_lightgbm/impl/__init__.py

```diff
@@ -79,7 +94,79 @@ def __init__(self, run: 'neptune.Run', base_namespace=''):
+            channel_name = 'foo/{}/foo2/{}_{}'.format(self._base_namespace, name, loss_name)
             self._run[channel_name].log(loss_value, step=env.iteration)
 
+
+def create_booster_summary(
+        booster: Union[lgb.Booster, lgb.sklearn.LGBMModel],
+        log_importances: bool = True,
+        max_num_features: int = 10,
+        list_trees: list = None,
+        log_trees_as_dataframe: bool = True,  # works only for lgb.Booster
+        log_pickled_booster: bool = True,
+        log_trees: bool = False,  # requires graphviz
+        log_confusion_matrix: bool = False,  # requires scikit-plot
+        y_true: np.ndarray = None,
+        y_pred: np.ndarray = None,
+):
+    results_dict = {}
+    visuals_path = "visualizations/"
+    if log_importances:
+        split_plot = lgb.plot_importance(
+            booster,
+            importance_type="split",
+            title="Feature importance (split)",
+            max_num_features=max_num_features
+        )
+        gain_plot = lgb.plot_importance(
+            booster,
+            importance_type="gain",
+            title="Feature importance (gain)",
+            max_num_features=max_num_features
+        )
+        results_dict["{}feature_importances/split".format(visuals_path)] \
+            = neptune.types.File.as_image(split_plot.figure)
+        results_dict["{}feature_importances/gain".format(visuals_path)] \
+            = neptune.types.File.as_image(gain_plot.figure)
+
+    # TODO: Your file is larger than 15MB. Neptune supports logging files in-memory objects smaller than 15MB.
+    #       Resize or increase compression of this object
+
+    if log_trees:
+        try:
+            trees_series = []
+            for i in list_trees:
+                digraph = lgb.create_tree_digraph(booster, tree_index=i, show_info='data_percentage')
+                _, ax = plt.subplots(1, 1)
+                s = BytesIO()
+                s.write(digraph.pipe(format='png'))
+                s.seek(0)
+                ax.imshow(image.imread(s))
+                ax.axis('off')
+                trees_series.append(neptune.types.File.as_image(ax.figure))
+            results_dict["{}trees".format(visuals_path)] = neptune.types.FileSeries(trees_series)
+        except GraphvizExecutableNotFound:
+            # TODO: link to Neptune docs section where will be such info: https://graphviz.org/download/
+            log.warning(f"Trees won't be logged. To make it work install graphviz library on your OS.")
+
+    if log_trees_as_dataframe:
+        if isinstance(booster, lgb.Booster):
+            df = booster.trees_to_dataframe()
+            results_dict["trees_as_dataframe"] = neptune.types.File.as_html(df)
+        else:
+            # TODO: log here and don't perform action, or detect it when calculation starts and raise an exception
+            log.warning("Trees won't be logged as dataframe. `booster` must be instance of `lightgbm.Booster` class.")
```

Comment:
`log_trees_as_dataframe` flag can be set only if `booster` of type `Booster` is passed.

Is this action ok, or we should raise exception when user invokes this function with wrong configuration setup.

Reply:
* Your implementation looks good.
* For larger models 15MB limit can happen. (discussed somewhere above.)